internal/services/v1/lookupwatch.go

```diff
@@ -0,0 +1,303 @@
+					},
+					SubjectRelation: &core.RelationReference{
+						Namespace: req.SubjectObjectType,
+						// TODO: Name tag as optional whereas dispatchLookupSubject require this parameter
+						Relation: req.OptionalSubjectRelation,
+					},
+				},
+				lsStream,
+			)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	//CALL ReachableResources
+	var resources []string
+	rrStream := dispatchpkg.NewHandlingDispatchStream(*ctx, func(result *dispatchv1.DispatchReachableResourcesResponse) error {
+		resources = append(resources, result.Resource.ResourceIds...)
+		return nil
+	})
+	var subjectRelation = update.Tuple.ResourceAndRelation.Relation
+	if req.ResourceObjectType == update.Tuple.ResourceAndRelation.Namespace && req.Permission != update.Tuple.ResourceAndRelation.Relation {
+		// Arrow resolution
+		reader := ds.SnapshotReader(*atRevision)
+		_, typeSystem, err := namespace.ReadNamespaceAndTypes((*stream).Context(), update.Tuple.ResourceAndRelation.Namespace, reader)
+		if err != nil {
+			return err
+		}
+		subjectRelations, err := typeSystem.SearchComputedUsersetRelations(update.Tuple.ResourceAndRelation.Relation)
+		if err != nil {
+			return err
+		}
+		for _, relation := range subjectRelations {
+			if relation == req.Permission {
+				subjectRelation = relation
+				break
+			}
+		}
+
+	}
+	err := lw.dispatch.DispatchReachableResources(
+		&dispatchv1.DispatchReachableResourcesRequest{
+			Metadata: &dispatchv1.ResolverMeta{
+				AtRevision:     atRevision.String(),
+				DepthRemaining: lw.defaultDepth,
+			},
+			ResourceRelation: &core.RelationReference{
+				Namespace: req.ResourceObjectType,
+				Relation:  req.Permission,
+			},
+			SubjectIds: []string{update.Tuple.ResourceAndRelation.ObjectId},
+			SubjectRelation: &core.RelationReference{
+				Namespace: update.Tuple.ResourceAndRelation.Namespace,
+				Relation:  subjectRelation,
+			},
+		},
+		rrStream,
+	)
+	if err != nil {
+		return err
+	}
+
+	//CROSS JOIN
+	permissionUpdates := []*v1lookupwatch.PermissionUpdate{}
+	for _, subject := range subjects {
+		for _, resource := range resources {
+			//CALL CHECK PERMISSION
+			permission, err := lw.dispatch.DispatchCheck(*ctx, &dispatchv1.DispatchCheckRequest{
+				Metadata: &dispatchv1.ResolverMeta{
+					AtRevision:     atRevision.String(),
+					DepthRemaining: lw.defaultDepth,
+				},
+				ResourceAndRelation: &core.ObjectAndRelation{
+					Namespace: req.ResourceObjectType,
+					ObjectId:  resource,
+					Relation:  req.Permission,
+				},
+				Subject: &core.ObjectAndRelation{
+					Namespace: req.SubjectObjectType,
+					ObjectId:  subject,
+					Relation:  graph.Ellipsis,
+				},
+			})
+			if err != nil {
+				return err
+			}
+
+			permissionUpdates = append(permissionUpdates, &v1lookupwatch.PermissionUpdate{
+				Subject: &v1.SubjectReference{
+					Object: &v1.ObjectReference{
+						ObjectType: req.SubjectObjectType,
+						ObjectId:   subject,
+					},
+					OptionalRelation: "", //TODO ?
```

Comment:
The request should be specifying this subject relation and using it here

Reply:
`optional_subject_relation` is what its called in the request

Reply:
done, changed with

```
OptionalRelation: req.OptionalSubjectRelation,
```

thanks !